### ISSUE TITLE
Thanks for the script!, seems to fail on linux-image-5.10.0-1029-oem

### ISSUE THREAD

**Issue by mtomlinson** [stukev/XPS-17-9700-Ubuntu-Soundfix#3]
I have used your script and it worked very well for me, thank you.

I have just noticed that with the latest upgrade of the kernel to linux-image-5.10.0-1029-oem the microphone seems to fail.
If I however simply run the older linux-image-5.6.0-1056-oem everything works fine.

any idea on why this might occur?

**Comment by stukev** [comment]
Have you checked your ubuntu sound settings? Is the microphone showing up there? Also try checking your alsa/alsamixer/pulseaudio settings too.

If it's still broken, I'm afraid it's probably the newer kernel's fault. Either they flat out broke compatibility or it might require a newer driver version than is present in my script. If you'd like to tinker, you could check if there's new versions of sof-tplg/sof-soundwire.

**Comment by mtomlinson** [comment]
Thanks, I have done some further digging and found these...
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1921632
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1912673﻿

Look like it is not far off being fixed (fingers crossed) I have decided to
wait it out a while.

Thanks for your help

**Comment by stukev** [comment]
Thanks for letting me know, that's good news indeed. Have you already tried the new Kernel suggested here? https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1912673/comments/24
I'm going to amend the project description and list the working kernels for now.

**Comment by Pozadr** [comment]
FYI 5.10.0-1034-oem has still internal-microphone issue with and without script.

**Comment by stukev** [comment]
Thanks for the report Pozadr, I'll update the readme.
I am currently running 5.10.0-1033-oem and it also seems to have the microphone issue. If anyone finds a fix please let me know.

**Comment by mtomlinson** [comment]
Hey, just an update, 5.11.0-25-generic works as per the comment in https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1912673

**Comment by stukev** [comment]
Thanks for your report. Does the microphone work too? I'll update the table of supported Kernels.

**Comment by mtomlinson** [comment]
Yes sorry to be clear everything works, but this is without the script, the issue has been resolved
